ambry-clustermap/src/test/java/com.github.ambry.clustermap/MockClusterMap.java

```diff
@@ -116,12 +121,20 @@ public MockClusterMap(boolean enableSSLPorts, int numNodes, int numMountPointsPe
     long partitionId = 0;
     for (int i = 0; i < dataNodes.get(0).getMountPaths().size(); i++) {
       for (int j = 0; j < numStoresPerMountPoint; j++) {
-        // TODO: allow partition class to be set for testing
         PartitionId id = new MockPartitionId(partitionId, DEFAULT_PARTITION_CLASS, dataNodes, i);
         partitions.put(partitionId, id);
         partitionId++;
       }
     }
+    if (numNodes > 3 && dcToDataNodes.get(localDatacenterName).size() > 2) {
+      // create one "special partition" that has 3 replicas in the local datacenter (as configured on startup) and
+      // 2 everywhere else
+      List<MockDataNodeId> nodeIds = new ArrayList<>();
+      dcToDataNodes.forEach(
+          (s, mockDataNodeIds) -> nodeIds.addAll(mockDataNodeIds.subList(0, localDatacenterName.equals(s) ? 3 : 2)));
+      PartitionId id = new MockPartitionId(partitionId, SPECIAL_PARTITION_CLASS, nodeIds, 0);
+      partitions.put(partitionId, id);
```

Comment:
In the statement `PartitionId id = new MockPartitionId(partitionId, SPECIAL_PARTITION_CLASS, nodeIds, 0);`,  a new `partitionId` is created and mounted on `mountPathIndexToUse = 0`.

Reply:
It does - It doesn't look like any of the tests care but I'll take a look at trying to maintain that invariant.

On the flip side, you can say that the callers want `numStoresPerMountPoint` "default" stores per mount point. Let me think over this.

Reply:
Done - changed docs to reflect all the changes